### ISSUE TITLE
“nix-env --upgrade” says “Unsupported Nix system: aarch64-darwin”

### ISSUE THREAD

**Issue by bdesham** [NixOS/nixpkgs#274817]
### Describe the bug

When I run

    nix-env --upgrade --dry-run

I get the following output:

```console
$ nix-env --upgrade --dry-run    
(dry run; not doing anything)
error:
       … while trying to find an upgrade for 'direnv-2.33.0'

       … in the right operand of the update (//) operator

         at /nix/store/8nklilzxgxvahmbcrimkpr72gldg10la-nixpkgs/nixpkgs/pkgs/stdenv/generic/check-meta.nix:479:5:

          478|     }
          479|     // attrs.meta or { }
             |     ^
          480|     # Fill `meta.position` to identify the source location of the package.

       … in the left operand of the update (//) operator

         at /nix/store/8nklilzxgxvahmbcrimkpr72gldg10la-nixpkgs/nixpkgs/pkgs/stdenv/generic/check-meta.nix:481:5:

          480|     # Fill `meta.position` to identify the source location of the package.
          481|     // optionalAttrs (pos != null) {
             |     ^
          482|       position = pos.file + ":" + toString pos.line;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Unsupported Nix system: aarch64-darwin
```

Metadata:

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.7.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.19.2`
 - channels(bdesham): `"nixpkgs"`
 - channels(root): `"nixpkgs-22.05pre348320.cc68710784f"`
 - nixpkgs: `/Users/bdesham/.nix-defexpr/channels/nixpkgs`
```

I tried to roll back to an environment with an earlier version of Nix, but I got essentially the same result:

```console
$ nix-env --upgrade --dry-run
(dry run; not doing anything)
error:
       … while trying to find an upgrade for 'sqlite-3.43.2'

       … in the right operand of the update (//) operator

         at /nix/store/8nklilzxgxvahmbcrimkpr72gldg10la-nixpkgs/nixpkgs/pkgs/stdenv/generic/check-meta.nix:479:5:

          478|     }
          479|     // attrs.meta or { }
             |     ^
          480|     # Fill `meta.position` to identify the source location of the package.

       … in the left operand of the update (//) operator

         at /nix/store/8nklilzxgxvahmbcrimkpr72gldg10la-nixpkgs/nixpkgs/pkgs/stdenv/generic/check-meta.nix:481:5:

          480|     # Fill `meta.position` to identify the source location of the package.
          481|     // optionalAttrs (pos != null) {
             |     ^
          482|       position = pos.file + ":" + toString pos.line;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Unsupported Nix system: aarch64-darwin
```

The metadata for that environment was:

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.7.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - channels(bdesham): `"nixpkgs"`
 - channels(root): `"nixpkgs-22.05pre348320.cc68710784f"`
 - nixpkgs: `/Users/bdesham/.nix-defexpr/channels/nixpkgs`
```

**Comment by thebitstick** [comment]
Same here on
```shell
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.2.0, macOS 14.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.11.1`
 - channels(thebitstick): `"nixpkgs"`
 - channels(root): `""`
 - nixpkgs: `/Users/thebitstick/.nix-defexpr/channels/nixpkgs`
 ```

**Comment by philipmw** [comment]
I was experiencing this on an existing macOS installation. Re-installing Nix (via the official install script) solved the problem.

**Comment by Bak-Jin-Hyeong** [comment]
In my case, the same symptom repeats even if I uninstall and reinstall nix.

```sh
$ nix-env --upgrade --dry-run --show-trace
(dry run; not doing anything)
error:
       … while trying to find an upgrade for 'zoxide-unstable-2023-11-20'

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:530:10:

          529|
          530|   meta = checkMeta.commonMeta { inherit validity attrs pos references; };
             |          ^
          531|   validity = checkMeta.assertValidity { inherit meta attrs; };

       … while calling 'commonMeta'

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/pkgs/stdenv/generic/check-meta.nix:452:16:

          451|   #   validity = checkMeta.assertValidity { inherit meta attrs; };
          452|   commonMeta = { validity, attrs, pos ? null, references ? [ ] }:
             |                ^
          453|     let

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/pkgs/top-level/all-packages.nix:7349:21:

         7348|   cudaPackages_11_8 = callPackage ./cuda-packages.nix { cudaVersion = "11.8"; };
         7349|   cudaPackages_11 = recurseIntoAttrs cudaPackages_11_8;
             |                     ^
         7350|

       … while calling 'recurseIntoAttrs'

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/attrsets.nix:1169:5:

         1168|     # An attribute set to scan for derivations.
         1169|     attrs:
             |     ^
         1170|     attrs // { recurseForDerivations = true; };

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/pkgs/top-level/all-packages.nix:7348:23:

         7347|   cudaPackages_11_7 = callPackage ./cuda-packages.nix { cudaVersion = "11.7"; };
         7348|   cudaPackages_11_8 = callPackage ./cuda-packages.nix { cudaVersion = "11.8"; };
             |                       ^
         7349|   cudaPackages_11 = recurseIntoAttrs cudaPackages_11_8;

       … while calling 'callPackageWith'

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/customisation.nix:153:35:

          152|   */
          153|   callPackageWith = autoArgs: fn: args:
             |                                   ^
          154|     let

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/trivial.nix:440:7:

          439|     { # TODO: Should we add call-time "type" checking like built in?
          440|       __functor = self: f;
             |       ^
          441|       __functionArgs = args;

       … while calling anonymous lambda

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/customisation.nix:96:17:

           95|     in
           96|     mirrorArgs (origArgs:
             |                 ^
           97|     let

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/customisation.nix:98:16:

           97|     let
           98|       result = f origArgs;
             |                ^
           99|

       … while calling anonymous lambda

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/pkgs/top-level/cuda-packages.nix:23:1:

           22| # I've (@connorbaker) attempted to do that, though I'm unsure of how this will interact with overrides.
           23| {
             | ^
           24|   callPackage,

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/pkgs/top-level/cuda-packages.nix:118:18:

          117|
          118|   cudaPackages = customisation.makeScope newScope (
             |                  ^
          119|     fixedPoints.extends composedExtension passthruFunction

       … while calling 'makeScope'

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/customisation.nix:317:25:

          316|   */
          317|   makeScope = newScope: f:
             |                         ^
          318|     let self = f self // {

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/customisation.nix:318:16:

          317|   makeScope = newScope: f:
          318|     let self = f self // {
             |                ^
          319|           newScope = scope: newScope (self // scope);

       … while calling 'extends'

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/fixed-points.nix:141:24:

          140|   */
          141|   extends = f: rattrs: self: let super = rattrs self; in super // f self super;
             |                        ^
          142|

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/fixed-points.nix:141:67:

          140|   */
          141|   extends = f: rattrs: self: let super = rattrs self; in super // f self super;
             |                                                                   ^
          142|

       … while calling 'composeExtensions'

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/fixed-points.nix:149:18:

          148|   composeExtensions =
          149|     f: g: final: prev:
             |                  ^
          150|       let fApplied = f final prev;

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/fixed-points.nix:152:22:

          151|           prev' = prev // fApplied;
          152|       in fApplied // g final prev';
             |                      ^
          153|

       … while calling 'composeExtensions'

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/fixed-points.nix:149:18:

          148|   composeExtensions =
          149|     f: g: final: prev:
             |                  ^
          150|       let fApplied = f final prev;

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/fixed-points.nix:152:22:

          151|           prev' = prev // fApplied;
          152|       in fApplied // g final prev';
             |                      ^
          153|

       … while calling 'composeExtensions'

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/fixed-points.nix:149:18:

          148|   composeExtensions =
          149|     f: g: final: prev:
             |                  ^
          150|       let fApplied = f final prev;

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/fixed-points.nix:152:22:

          151|           prev' = prev // fApplied;
          152|       in fApplied // g final prev';
             |                      ^
          153|

       … while calling 'composeExtensions'

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/fixed-points.nix:149:18:

          148|   composeExtensions =
          149|     f: g: final: prev:
             |                  ^
          150|       let fApplied = f final prev;

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/fixed-points.nix:150:22:

          149|     f: g: final: prev:
          150|       let fApplied = f final prev;
             |                      ^
          151|           prev' = prev // fApplied;

       … while calling 'extension'

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/pkgs/development/cuda-modules/generic-builders/multiplex.nix:104:12:

          103|   extension =
          104|     final: _:
             |            ^
          105|     let

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/pkgs/development/cuda-modules/generic-builders/multiplex.nix:86:36:

           85|   # newestToOldestSupportedPackage :: List (AttrSet Packages)
           86|   newestToOldestSupportedPackage = lists.reverseList supportedPackages;
             |                                    ^
           87|

       … while calling 'reverseList'

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/lib/lists.nix:510:17:

          509|   */
          510|   reverseList = xs:
             |                 ^
          511|     let l = length xs; in genList (n: elemAt xs (l - n - 1)) l;

       … from call site

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/pkgs/development/cuda-modules/generic-builders/multiplex.nix:79:16:

           78|   # Get all of the packages for our given platform.
           79|   redistArch = flags.getRedistArch hostPlatform.system;
             |                ^
           80|

       … while calling 'getRedistArch'

         at /nix/store/55swk71whn1hppd70x14ld9k6f0413dw-nixpkgs/nixpkgs/pkgs/development/cuda-modules/flags.nix:136:5:

          135|   getRedistArch =
          136|     nixSystem:
             |     ^
          137|     if nixSystem == "aarch64-linux" then

       error: Unsupported Nix system: aarch64-darwin

```


```sh
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.2.0, macOS 14.2.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.19.2`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by philipmw** [comment]
In my case, I uninstalled all packages, then upgraded/replaced Nix, then started reinstalling packages.

**Comment by lejoteux** [comment]
I'm having the same issue. I think it occurs during updates but not when installing new packages. I can install packages without any issues but I cannot update them with `nix-env -u`.

**Comment by ink-splatters** [comment]
It's consistent with what's happening on my side. I've made kind-of workaround (? sorry still consider myself nix noob).

sys info:

```shell
root# nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.2.0, macOS 14.2.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.19.2`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/var/root/.nix-defexpr/channels/nixpkgs`
```

```shell
root# nix-shell -q
```
<details>
<summary>result</summary>

```shell
bandwhich-0.21.1
bash-interactive-5.2-p21
bat-0.24.0
btop-1.2.13
cachix-1.6.1
chafa-1.12.5
charm-0.12.6
colima-0.6.7
delta-0.16.5
docker-24.0.5
du-dust-0.8.6
dua-2.23.0
duf-0.8.1
eza-0.17.0
fclones-0.34.0
fd-9.0.0
fend-1.3.3
ffsend-0.2.76
flyctl-0.1.134
fzf-0.44.1
git-lfs-3.4.1
git-minimal-2.42.0
gitoxide-0.31.1
gitui-0.24.3
glow-1.5.1
gnumake-4.4.1
gnupg-2.4.1
go-1.21.5
gptman-1.0.2
inetutils-2.4
inlyne-0.3.2
ipatool-2.1.3
jiq-0.7.2
jq-1.7
jwt-hack-1.1.2
kitty-0.31.0
lemmeknow-0.8.0
lldb-17.0.6
melt-0.6.0
micromamba-1.5.4
nextdns-1.41.0
ngrep-1.47
nix-2.19.2
nix-melt-0.1.2
nix-tree-0.3.1
nmap-7.94
nomino-1.3.3
nss-cacert-3.95
oq-1.3.4
pastel-0.9.0
procs-0.14.4
protonmail-bridge-3.7.1
pspg-5.8.1
pv-1.8.5
rectangle-0.75
rustup-1.26.0
shellharden-4.3.0
shfmt-3.7.0
sipcalc-1.1.6
slides-0.9.0
tldr-1.6.0
tv-0.7.0
viu-1.4.0
vivid-0.9.0
xh-0.20.1
xxh-0.8.12
xxHash-0.8.2
zoxide-unstable-2023-11-20
zstd-1.5.5
```
</details>

everything seems to work if using `nix-env --upgrade --attr nixpkgs.<versionless package name> --attr ...`


so I've hacked some shell magic which prepares the input for `nix-env --upgrade`, stripping off the version info (yes it's stupid but I dunno of the better way:) ).

Also it's hardcoded to patch things like:

- _bash-interactive_ -> _**bashInteractive**_
- _git-minimal_ -> **_gitMinimal_**
- ...

```shell
nix-shell -p ripgrep --run sh < <(cat <<'EOF'
nix-env -q | rg -o '^([a-zA-Z-]+)[.-](?:[\d.-]+)*' --replace '--attr nixpkgs.$1' | sed -e 's/-i/I/g;s/-min/Min/g;s/nss-//g;s/-unstable//g'
EOF
```

<details>
<summary>result</summary>

```shell
--attr nixpkgs.bandwhich
--attr nixpkgs.bashInteractive
--attr nixpkgs.bat
--attr nixpkgs.btop
--attr nixpkgs.cachix
--attr nixpkgs.chafa
--attr nixpkgs.charm
--attr nixpkgs.colima
--attr nixpkgs.delta
--attr nixpkgs.docker
--attr nixpkgs.du-dust
--attr nixpkgs.dua
--attr nixpkgs.duf
--attr nixpkgs.eza
--attr nixpkgs.fclones
--attr nixpkgs.fd
--attr nixpkgs.fend
--attr nixpkgs.ffsend
--attr nixpkgs.flyctl
--attr nixpkgs.fzf
--attr nixpkgs.git-lfs
--attr nixpkgs.gitMinimal
--attr nixpkgs.gitoxide
--attr nixpkgs.gitui
--attr nixpkgs.glow
--attr nixpkgs.gnumake
--attr nixpkgs.gnupg
--attr nixpkgs.go
--attr nixpkgs.gptman
--attr nixpkgs.inetutils
--attr nixpkgs.inlyne
--attr nixpkgs.ipatool
--attr nixpkgs.jiq
--attr nixpkgs.jq
--attr nixpkgs.jwt-hack
--attr nixpkgs.kitty
--attr nixpkgs.lemmeknow
--attr nixpkgs.lldb
--attr nixpkgs.melt
--attr nixpkgs.micromamba
--attr nixpkgs.nextdns
--attr nixpkgs.ngrep
--attr nixpkgs.nix
--attr nixpkgs.nix-melt
--attr nixpkgs.nix-tree
--attr nixpkgs.nmap
--attr nixpkgs.nomino
--attr nixpkgs.cacert
--attr nixpkgs.oq
--attr nixpkgs.pastel
--attr nixpkgs.procs
--attr nixpkgs.protonmail-bridge
--attr nixpkgs.pspg
--attr nixpkgs.pv
--attr nixpkgs.rectangle
--attr nixpkgs.rustup
--attr nixpkgs.shellharden
--attr nixpkgs.shfmt
--attr nixpkgs.sipcalc
--attr nixpkgs.slides
--attr nixpkgs.tldr
--attr nixpkgs.tv
--attr nixpkgs.viu
--attr nixpkgs.vivid
--attr nixpkgs.xh
--attr nixpkgs.xxh
--attr nixpkgs.xxHash
--attr nixpkgs.zoxide
--attr nixpkgs.zstd
```

</details>


with final upgrade command looking like this:

```shell
nix-shell -p ripgrep gnused --run sh < <(cat <<'EOF'
nix-env -q | rg -o '^([a-zA-Z-]+)[.-](?:[\d.-]+)*' --replace '--attr nixpkgs.$1' | sed -e 's/-i/I/g;s/-min/Min/g;s/nss-//g;s/-unstable//g' | xargs nix-env --upgrade
EOF
```

it works fine (note `gnused`, it's needed because macOS `sed` doesn't support chaining, apparently)

-------------------

But: i found something strange, e.g. for `zstd` package:

```shell
nix-env --upgrade --attr nixpkgs.zstd
```

works as expected, but

```shell
nix-env --upgrade zstd
```

does something different:

```shell
upgrading 'zstd-1.5.5' to 'zstd-20230618-git'
this derivation will be built:
  /nix/store/dhzys44yfkcb19m3h96jgn4x7i791vmd-zstd-20230618-git.drv
...
BUILD FAILED: Unable to load foreign library (LIBZSTD).
  Error opening shared object "libzstd.dylib":
  dlopen(libzstd.dylib, 0x000A): tried: 'libzstd.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibzstd.dylib' (no such file), '/usr/lib/libzstd.dylib' (no such file, not in dyld cache), 'libzstd.dylib' (no such file).
error: builder for '/nix/store/dhzys44yfkcb19m3h96jgn4x7i791vmd-zstd-20230618-git.drv' failed with exit code 1
```

which results in build failure (but that's another story).

I guess this invocation is close to what `nix-env --upgrade` or `nix-env --upgrade '.*'` does?

Again, I might be misunderstanding things because of noobness, on the other hand it's not really obvious contract.

**Comment by onetom** [comment]
Same here, but it breaks on the `nss-cacert` package, after successfully running `sudo -i nix-channel --update`:

```
❯ sudo -i nix-channel --list  
nixpkgs https://nixos.org/channels/nixpkgs-unstable

❯ sudo -i nix-env -u  
error:
       … while trying to find an upgrade for 'nss-cacert-3.92'

       … in the right operand of the update (//) operator

         at /nix/store/j8sh7dkdp9lrlw6qhyj415zzy8vg2ds2-nixpkgs/nixpkgs/pkgs/stdenv/generic/check-meta.nix:479:5:

          478|     }
          479|     // attrs.meta or { }
             |     ^
          480|     # Fill `meta.position` to identify the source location of the package.

       … in the left operand of the update (//) operator

         at /nix/store/j8sh7dkdp9lrlw6qhyj415zzy8vg2ds2-nixpkgs/nixpkgs/pkgs/stdenv/generic/check-meta.nix:481:5:

          480|     # Fill `meta.position` to identify the source location of the package.
          481|     // optionalAttrs (pos != null) {
             |     ^
          482|       position = pos.file + ":" + toString pos.line;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Unsupported Nix system: aarch64-darwin
```

As recommended above, explicitly asking for the individual packages to upgrade did work:
```
❯ sudo -i nix-env --upgrade nss-cacert  
upgrading 'nss-cacert-3.92' to 'nss-cacert-3.95'
these 2 paths will be fetched (0.32 MiB download, 0.93 MiB unpacked):
  /nix/store/f7nd6i5yzspkrhmqilmzf886mxw7h4gr-nss-cacert-3.95-p11kit
  /nix/store/4f7kdya0mnkabmpa03lb007bwcw2c8p6-nss-cacert-3.95-unbundled
copying path '/nix/store/f7nd6i5yzspkrhmqilmzf886mxw7h4gr-nss-cacert-3.95-p11kit' from 'https://cache.nixos.org'...
copying path '/nix/store/4f7kdya0mnkabmpa03lb007bwcw2c8p6-nss-cacert-3.95-unbundled' from 'https://cache.nixos.org'...
building '/nix/store/4akp26vp3gnkysfwvgfdifx2m72pkms8-user-environment.drv'...

❯ sudo -i nix-env --upgrade nix
...
copying path '/nix/store/h3h900clwa2sri36pi4fwhzy2pq8p3iw-nix-2.19.2' from 'https://cache.nixos.org'...
copying path '/nix/store/hymkh0h91khxr6x8h2zgk28vhd63jr2q-nix-2.19.2-dev' from 'https://cache.nixos.org'...
building '/nix/store/h8lhvml2ns2yvwcgxrdc4gzfg6yfj05h-user-environment.drv'...
```

but upgrade on its own still fails with the same errors, even after these manual upgrades:
```
❯ sudo -i nix-env -u           
error:
       … while trying to find an upgrade for 'nss-cacert-3.95'
...
```

**Comment by ink-splatters** [comment]
@onetom yes, I guess it's a matter of bisecting some commit ~2 weeks ago or so? 
And fix it. Only if it's not a feature :)

**Comment by bdesham** [comment]
As far as I can tell, the problem was not present in nixpkgs commit 8a5b9ee7b7a2b38267c9481f5c629c015108ab0d, and it was present by commit 41ac0bd371618db6dd67fd952cc5b3d6a9955a15. I would bisect, but it’s not clear to me how you can point `nix-env` toward a specific commit. (According to [this page](https://nixos.org/guides/nix-pills/nix-search-paths.html), `nix-env` ignores `NIX_PATH`.)

**Comment by bdesham** [comment]
Over on #276413, @holymonson notes that the problems seems to have been introduced in 8e800cedaf24f5ad9717463b809b0beef7677000.

**Comment by apinter** [comment]
FWIW I just reinstalled `nix` and the upgrade worked just fine.

**Comment by ConnorBaker** [comment]
Should have been resolved by mid-January. Issue was caused by Darwin evaluating cudaPackages more deeply than I had expected and throwing as a result (stack trace in https://github.com/NixOS/nixpkgs/issues/274817#issuecomment-1868228718).
